tests/integration/upgrade/test_upgrade.py

```diff
@@ -31,12 +34,41 @@ async def test_build_and_deploy(ops_test: OpsTest) -> None:
     if app_name:
         return await check_or_scale_app(ops_test, app_name)
 
-    my_charm = await ops_test.build_charm(".")
-    await ops_test.model.deploy(my_charm, num_units=3)
-    await ops_test.model.wait_for_idle()
+    await ops_test.model.deploy("mongodb", channel="stable", num_units=3)
+    await ops_test.model.wait_for_idle(
+        apps=["mongodb"], status="active", timeout=1000, idle_period=120
+    )
+
+
+@pytest.mark.group(1)
+@pytest.mark.skip(reason="Must wait for stable channel to support upgrades before testing")
```

Comment:
Can we rely on `edge` so these tests get run?

Reply:
In theory yes, but we only support upgrades to `stable` revisions, I believe our tests should reflect that.

Reply:
yes  you are right! I will add a comment saying that we should use stable ASAP though :)